cassandra/connection.py

```diff
@@ -782,6 +785,20 @@ def _wrap_socket_from_context(self):
         self._socket = self.ssl_context.wrap_socket(self._socket, **ssl_options)
 
     def _initiate_connection(self, sockaddr):
+        if self.shard_id is not None:
+            # TODO: move to constructor
```

Comment:
TODO or not TODO ? :)

Reply:
I thought it might be better to have the list of source ports created one per shard, but them I've turned it into a generator, and maybe that would be enough.

I think maybe randomizing it a bit, might help, i.e. on a long running machine, maybe lots of the lower ports are already taken..., but it's just a guess